### ISSUE TITLE
slow upload speed

### ISSUE THREAD

**Issue by newcron** [dasuchin/grunt-ssh-deploy#46]
hi guys,
maybe you do have an idea: 
I have a rather big release (150mb, gzipped with max compression). If I upload it from the shell via SCP, I get ~1.5 mb/sec. but if I upload it with grunt-ssh-deploy, I only get ~100 kb/sec which makes the deployment painfully long... 

have you ever experienced something like this? 

this is my config: 

```
 environments: {
            options: {
                local_path: "atlas-release.tar.gz",
                deploy_path: "~/releases"
            },
            aws_test: {
                options: {
                    host: "xxx",
                    username: "devenv",
                    port: 2222,
                    agent: process.env.SSH_AUTH_SOCK,
                    debug: true,
                    releases_to_keep: 1,
                    after_deploy: "cd ~/releases/current && tar -zxf release.tar.gz"
                }
            }
        }
```

**Comment by tamitutor** [comment]
+1

**Comment by aborovsky** [comment]
+1
Пн, 14 сент. 2015 г. в 21:54, Tami Wright <notifications@github.com>:

**Comment by tamitutor** [comment]
Until this is fixed, I'm reverting to [grunt-ssh] (https://github.com/chuckmo/grunt-ssh) and implementing what's outlined in this [blog post] (http://tech.toptable.co.uk/blog/2013/08/08/grunt-your-deployments-too/).  If I find the time to create a patch for this package I will--and post a pull-request.

**Comment by dasuchin** [comment]
Have you tried using the zip option? The plugin itself uses scp, so I'm not sure why it would be any slower than just running scp yourself.

**Comment by tamitutor** [comment]
Yes, I did try zip and it was still extremely slow...a 12.9mb zip file wasn't uploaded after an hour.  I am using a private key instead of a password if that helps at all.  (A private key doesn't seem to affect how quickly *_deploy hooks--they are executed plenty fast...)

**Comment by tamitutor** [comment]
After moving everything over to use what I mentioned I found in the blog post (see comment before last) I had the same problems...it appears to be a sftp issue--possibly.  I'm now using grunt-exec with a "scp" command to get my file on the server--and it's faster...MUCH faster.

**Comment by newcron** [comment]
it's definitely not about zipping content. 
I measured the bytes that were sent out by my PC to the network. Using grunt-ssh-deploy, only 100kb/sec are sent out to the internet, using scp directly, then it's 15x more. 

If zipping the contents would have been the different, the bytes/sec should have been identical while the number of total bytes would differ.
